### PR TITLE
Display registration keys in university profile page

### PR DESCRIPTION
**Pull request by pedbr**
### User Story

```
As university user,
In order to enable my research groups to sign up,
I want to see the registration keys associated with my paid subscription on my profile page.
```
### Feature description
https://www.pivotaltracker.com/story/show/168108135

### Description of PR
PR introduces a University User redirect from the payment form to the user profile page (if payment is succeeded), and here the Research Group Keys will be displayed. We are making a GET request to the API to fetch that particular user's keys. The response handling is perfectly aligned with what is being sent to us from the backend.

### Screenshots
Screen Shot 2019-08-28 at 00 34 48

### What did we learn?
The importance of having the front and backend aligned so that the requests are well made, and the responses handled properly.

### CODE REVIEW

File: src/components/NavBar.js [hunk 1]
```diff
@@ -48,8 +49,16 @@ class NavBar extends Component {
...
         </>
       )
+    }
 
+    if (this.props.currentUser.isSignedIn === true) {
+      userProfileButton = (
+        <>
+          <Menu.Item as={NavLink} to="/profile" id="profile-button">Profile</Menu.Item>
+        </>
+      )
```
Comment (julieahlquist) [inline src/components/NavBar.js:61]: Can all user roles see their profile or should this be re-written to only subscribed university (as on line 31-32)

Comment (pedbr) [reply src/components/NavBar.js:61]: All users will access the profile component, then the profile component content will vary depending on the user role

File: cypress/integration/universityCanSeeRegistrationKeys.spec.js [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    cy.get("#registration-keys-title").should("exist");
+    cy.contains("b3tDSWucTHtzWLNNGFdgagip");
+    cy.contains("U9YT35JejpfKDy1vfuWTVLyb");
+    cy.contains("dWCWyfnNn4DVFEHzAAz7b6pw");
+    cy.contains("6evmpvxxJmHxvWdNZzZYKa4i");
+    cy.contains("fGygSg6Fxjav78tKEbLzS3TB");
+  })
+
+  it("Unsubscribed University cannot see registration keys on profile", () => {
+    cy.university_login("harvard@mail.com", "password");
```
Comment (julieahlquist) [inline cypress/integration/universityCanSeeRegistrationKeys.spec.js]: please follow conventions
university email = harvard@harvard.edu

File: cypress/fixtures/successful_subscribed_university_login.json [hunk 3]
```diff
@@ -0,0 +1,10 @@
+{
+  "data": {
+    "id": 1,
+    "name": "Harvard",
+    "email": "harvard@mail.com",
```
Comment (julieahlquist) [inline cypress/fixtures/successful_subscribed_university_login.json]: mind the email here too please

File: cypress/integration/universityCanSeeRegistrationKeys.spec.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  beforeEach(() => {
+    cy.server();
+    cy.route({
+      method: "GET",
+      url: "http://localhost:3000/api/v0/articles",
+      response: "fixture:articles.json",
+      status: 200
+    });
+  });
+
```
Comment (julieahlquist) [inline cypress/integration/universityCanSeeRegistrationKeys.spec.js:12]: you dont need to test for the articles to be displayed in this feature

File: cypress/integration/universityCanSeeRegistrationKeys.spec.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  it("Subscribed University can access profile page and see registration keys", () => {
+    cy.route({
+      method: "GET",
+      url: "http://localhost:3000/api/v0/users/1",
+      response: "fixture:university_registration_keys.json",
+      status: 200
+    });
+    cy.subscribed_university_login("harvard@mail.com", "password");
+    cy.get("#subscribe-button").should("not.exist");
```
Comment (julieahlquist) [inline cypress/integration/universityCanSeeRegistrationKeys.spec.js:21]: email here too

File: cypress/integration/universityCanSeeRegistrationKeys.spec.js (hunk 2 above)
Comment (pedbr) [reply cypress/integration/universityCanSeeRegistrationKeys.spec.js]: fixed!

File: cypress/fixtures/successful_subscribed_university_login.json (hunk 3 above)
Comment (pedbr) [reply cypress/fixtures/successful_subscribed_university_login.json]: fixed!

File: cypress/integration/universityCanSeeRegistrationKeys.spec.js (hunk 5 above)
Comment (pedbr) [reply cypress/integration/universityCanSeeRegistrationKeys.spec.js:21]: fixed!

File: cypress/integration/universityCanSeeRegistrationKeys.spec.js (hunk 4 above)
Comment (pedbr) [reply cypress/integration/universityCanSeeRegistrationKeys.spec.js:12]: It's not that the purpose it to test the articles being displayed, but when you start the server the first component rendered is the homepage, that will always look for articles to display, and if you don't define the route with the fixture with the stubbed response, we will get an aborted request in the test, and we don't want that

**Pull request merged**